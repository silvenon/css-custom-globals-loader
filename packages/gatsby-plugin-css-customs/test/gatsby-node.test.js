const TestUtils = require('./utils')
const { loaderSerializer } = require('./serializers')

const {
  onCreateWebpackConfig: applyPostcss,
} = require('gatsby-plugin-postcss/gatsby-node')
const {
  onCreateWebpackConfig: applyCssModulesFlowTypes,
} = require('gatsby-plugin-css-modules-flow-types/gatsby-node')
const { onCreateWebpackConfig: applyCssCustoms } = require('../gatsby-node')

expect.addSnapshotSerializer(loaderSerializer)

it('applies the loader', () => {
  const utils = new TestUtils()
  applyPostcss(...utils.mockParams())
  applyCssCustoms(...utils.mockParams())
  expect(utils.getCssRules()).toMatchSnapshot()
})

it('composes well with similar plugins', () => {
  const utils = new TestUtils()
  applyPostcss(...utils.mockParams())
  applyCssModulesFlowTypes(...utils.mockParams())
  applyCssCustoms(...utils.mockParams())
  expect(utils.getCssRules()).toMatchSnapshot()
})

describe('SSR', () => {
  it('exports only locals', () => {
    const utils = new TestUtils({ stage: 'html' })
    applyPostcss(...utils.mockParams())
    applyCssCustoms(...utils.mockParams())
    expect(utils.getCssRules()).toMatchSnapshot()
  })
})
