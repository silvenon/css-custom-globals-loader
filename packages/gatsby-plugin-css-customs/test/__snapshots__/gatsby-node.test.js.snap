// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SSR exports only locals 1`] = `
Array [
  Object {
    "test": /\\\\\\.module\\\\\\.css\\$/,
    "use": Array [
      Object {
        "loader": "css-customs-loader/index.js",
        "options": Object {
          "onlyLocals": true,
        },
      },
      Object {
        "loader": "css-loader/locals",
        "options": Object {},
      },
    ],
  },
  Object {
    "test": /\\\\\\.css\\$/,
    "use": Array [
      Object {
        "loader": "css-customs-loader/index.js",
        "options": Object {
          "onlyLocals": true,
        },
      },
      Object {
        "loader": "css-loader/locals",
        "options": Object {},
      },
    ],
  },
]
`;

exports[`applies the loader 1`] = `
Array [
  Object {
    "test": /\\\\\\.module\\\\\\.css\\$/,
    "use": Array [
      Object {
        "loader": "mini-css-extract-loader",
        "options": Object {
          "hmr": false,
        },
      },
      Object {
        "loader": "css-customs-loader/index.js",
        "options": Object {
          "onlyLocals": false,
        },
      },
      Object {
        "loader": "css-loader",
        "options": Object {
          "importLoaders": 1,
          "modules": true,
        },
      },
      Object {
        "loader": "postcss-loader/src/index.js",
        "options": Object {
          "sourceMap": true,
        },
      },
    ],
  },
  Object {
    "test": /\\\\\\.css\\$/,
    "use": Array [
      Object {
        "loader": "mini-css-extract-loader",
        "options": Object {},
      },
      Object {
        "loader": "css-customs-loader/index.js",
        "options": Object {
          "onlyLocals": false,
        },
      },
      Object {
        "loader": "css-loader",
        "options": Object {
          "importLoaders": 1,
        },
      },
      Object {
        "loader": "postcss-loader/src/index.js",
        "options": Object {
          "sourceMap": true,
        },
      },
    ],
  },
  Object {
    "test": /\\\\\\.module\\\\\\.css\\$/,
    "use": Array [
      Object {
        "loader": "style-loader",
        "options": Object {},
      },
      Object {
        "loader": "css-customs-loader/index.js",
        "options": Object {
          "onlyLocals": false,
        },
      },
      Object {
        "loader": "css-loader",
        "options": Object {
          "modules": true,
        },
      },
    ],
  },
  Object {
    "test": /\\\\\\.css\\$/,
    "use": Array [
      Object {
        "loader": "style-loader",
        "options": Object {},
      },
      Object {
        "loader": "css-customs-loader/index.js",
        "options": Object {
          "onlyLocals": false,
        },
      },
      Object {
        "loader": "css-loader",
        "options": Object {},
      },
    ],
  },
]
`;

exports[`composes well with similar plugins 1`] = `
Array [
  Object {
    "test": /\\\\\\.module\\\\\\.css\\$/,
    "use": Array [
      Object {
        "loader": "mini-css-extract-loader",
        "options": Object {
          "hmr": false,
        },
      },
      Object {
        "loader": "css-modules-flow-types-loader/dist/index.js",
        "options": Object {},
      },
      Object {
        "loader": "css-customs-loader/index.js",
        "options": Object {
          "onlyLocals": false,
        },
      },
      Object {
        "loader": "css-loader",
        "options": Object {
          "importLoaders": 1,
          "modules": true,
        },
      },
      Object {
        "loader": "postcss-loader/src/index.js",
        "options": Object {
          "sourceMap": true,
        },
      },
    ],
  },
  Object {
    "test": /\\\\\\.css\\$/,
    "use": Array [
      Object {
        "loader": "mini-css-extract-loader",
        "options": Object {},
      },
      Object {
        "loader": "css-customs-loader/index.js",
        "options": Object {
          "onlyLocals": false,
        },
      },
      Object {
        "loader": "css-loader",
        "options": Object {
          "importLoaders": 1,
        },
      },
      Object {
        "loader": "postcss-loader/src/index.js",
        "options": Object {
          "sourceMap": true,
        },
      },
    ],
  },
  Object {
    "test": /\\\\\\.module\\\\\\.css\\$/,
    "use": Array [
      Object {
        "loader": "style-loader",
        "options": Object {},
      },
      Object {
        "loader": "css-customs-loader/index.js",
        "options": Object {
          "onlyLocals": false,
        },
      },
      Object {
        "loader": "css-loader",
        "options": Object {
          "modules": true,
        },
      },
    ],
  },
  Object {
    "test": /\\\\\\.css\\$/,
    "use": Array [
      Object {
        "loader": "style-loader",
        "options": Object {},
      },
      Object {
        "loader": "css-customs-loader/index.js",
        "options": Object {
          "onlyLocals": false,
        },
      },
      Object {
        "loader": "css-loader",
        "options": Object {},
      },
    ],
  },
]
`;
